thorlcr/activities/parse/thparseslave.cpp

```diff
@@ -65,7 +65,6 @@ class CParseSlaveActivity : public CSlaveActivity, implements IMatchedAction
     {
         ActivityTimer s(totalCycles, timeActivities);
         PARENT::start();
-        rowIter->first();
```

Comment:
I don't think this is sufficient.
There also should be a call to parser->reset() (see code in hthor), otherwise if the child query had a CHOOSEN() rowIter->isValid() would still be true.

Reply:
right, yes you're right and in fact that is initially what I change it too, but wondered by it worked in old builds.
As it is (in this commit ) it behaves as 5.4.10 did, which is okay if the input is exhausted isValid() becomes false and performMatch performs the reset, but as you say in a choosen case it will and would have (<=5.4.10 too) give the wrong results.
I'll add a reset call in stop()